### PR TITLE
chore(astro): improve example, add docs

### PR DESCRIPTION
**Pull request by juliusmarminge**

### CODE REVIEW

File: docs/src/pages/getting-started/_meta.json [hunk 1]
```diff
@@ -4,5 +4,5 @@
   "solid": "SolidStart",
   "svelte": "SvelteKit (Coming Soon)",
   "nuxt": "Nuxt.js (Coming Soon)",
-  "astro": "Astro (Coming Soon)"
+  "astro": "Astro"
```
Comment (t3dotgg) [inline docs/src/pages/getting-started/_meta.json:7]: Should be moved above the "coming soon" options

File: docs/src/pages/getting-started/astro.mdx [hunk 2]
```diff
@@ -1 +1,229 @@
...
+import EnvSetup from "./_shared/env-setup.mdx";
+import TailwindSetup from "./_shared/tailwind-setup.mdx";
+
+# Astro Setup
+
+You can use Astro with any of the frontend frameworks we support. This guide
+will walk you through setting it up using React, but the steps will be near
+identical for other frameworks too.
+
+<Callout type="info">
```
Comment (t3dotgg) [inline docs/src/pages/getting-started/astro.mdx:12]: Not in love with this. It's fine but I'd love for us to find a better place to visibly include "examples" in future

File: docs/src/pages/getting-started/astro.mdx [hunk 3]
```diff
@@ -1 +1,229 @@
...
+    }),
+} satisfies FileRouter;
+
+export type OurFileRouter = typeof ourFileRouter;
+```
+
+### Create a Astro API route using the FileRouter
+
+<Callout type="info">
+  File path here doesn't matter, you can serve this from any route. We recommend
```
Comment (t3dotgg) [inline docs/src/pages/getting-started/astro.mdx:71]: I would flip this to "We recommend serving the UploadThing endpoint out of /api/uploadthing, but you [can change the route if you like]" with a link to the custom endpoint doc

File: docs/src/pages/getting-started/astro.mdx [hunk 4]
```diff
@@ -1 +1,229 @@
...
+import type { OurFileRouter } from "~/server/uploadthing";
+
+export const UploadButton = generateUploadButton<OurFileRouter>();
+export const UploadDropzone = generateUploadDropzone<OurFileRouter>();
+```
+
+### Mount A Button And Upload!
+
+<Callout type="info">
+  Don't forget to add the `"use client;"` directive at the top of your file,
```
Comment (t3dotgg) [inline docs/src/pages/getting-started/astro.mdx:132]: Is this needed in Astro? Should we be telling them to use the client directives on the Astro side?

File: docs/src/pages/getting-started/astro.mdx (hunk 3 above)
Comment (juliusmarminge) [reply docs/src/pages/getting-started/astro.mdx:71]: just copied from next docs

File: docs/src/pages/getting-started/astro.mdx (hunk 4 above)
Comment (juliusmarminge) [reply docs/src/pages/getting-started/astro.mdx:132]: whoops nope

File: docs/src/pages/getting-started/astro.mdx (hunk 2 above)
Comment (juliusmarminge) [reply docs/src/pages/getting-started/astro.mdx:12]: agree - but let's do that in another PR where we deal with that explicitely for all doc/example combos

File: docs/src/pages/getting-started/astro.mdx (hunk 3 above)
Comment (juliusmarminge) [reply docs/src/pages/getting-started/astro.mdx:71]: i wont change it in this PR, do a follow-up changing all of them instead

**Pull request merged**